router/oauthResponseHandler/oauthResponseHandler.go

```diff
@@ -0,0 +1,546 @@
+	pkgLogger            logger.LoggerI
+	loggerNm             string
+	workspaceToken       string
+	isMultiWorkspace     bool
+	destAuthInfoMap      map[string]*AuthResponse // Stores information about account after refresh token/fetch token
+	accountLockMap       map[string]*sync.RWMutex // Lock used at account level to handle multiple refresh/fetch token requests
+	destLockMap          map[string]*sync.RWMutex // Lock used at destination level to handle multiple disable destination requests
+	lockMapMutex         *sync.RWMutex            // Lock used at destination level to handle multiple disable destination requests
+	refreshActiveMap     map[string]bool
+	disableDestActiveMap map[string]bool
+)
+
+const (
+	DISABLE_DEST                = "DISABLE_DESTINATION"
+	REFRESH_TOKEN               = "REFRESH_TOKEN"
+	INVALID_REFRESH_TOKEN_GRANT = "refresh_token_invalid_grant"
+)
+
+// The response from the transformer network layer will be sent with a output property(in case of an error)
+type ErrorOutput struct {
+	Output ErrorResponse `json:"output"`
+}
+
+// This struct represents the datastructure present in Transformer network layer Error builder
+type ErrorResponse struct {
+	Message             string                 `json:"message"`
+	Destination         map[string]interface{} `json:"destination"`
+	NetworkFailure      bool                   `json:"networkFailure"`
+	Status              int                    `json:"status"`
+	AuthErrorCategory   string                 `json:"authErrorCategory"`
+	AccessToken         string                 `json:"accessToken"`
+	StatTags            map[string]string      `json:"statTags"`
+	StatName            string                 `json:"statName"`
+	DestinationResponse map[string]interface{} `json:"destinationResponse"`
+}
+
+type RefreshTokenBodyParams struct {
+	HasExpired   bool   `json:"hasExpired"`
+	ExpiredToken string `json:"expiredToken"`
+}
+
+func Init() {
+	configBEURL = backendconfig.GetConfigBackendURL()
+	pkgLogger = logger.NewLogger().Child("router").Child("OAuthResponseHandler")
+	loggerNm = "OAuthResponseHandler"
+	workspaceToken = backendconfig.GetWorkspaceToken()
+	destAuthInfoMap = make(map[string]*AuthResponse)
+	accountLockMap = make(map[string]*sync.RWMutex)
+	destLockMap = make(map[string]*sync.RWMutex)
+	lockMapMutex = &sync.RWMutex{}
+	refreshActiveMap = make(map[string]bool)
+	disableDestActiveMap = make(map[string]bool)
+}
+
+func (authErrHandler *OAuthErrResHandler) Setup() {
+	authErrHandler.logger = pkgLogger
+	authErrHandler.tr = &http.Transport{}
+	//This timeout is kind of modifiable & it seemed like 10 mins for this is too much!
+	authErrHandler.client = &http.Client{}
+	authErrHandler.destLockMap = destLockMap
+	authErrHandler.accountLockMap = accountLockMap
+	authErrHandler.lockMapWMutex = lockMapMutex
+	authErrHandler.destAuthInfoMap = destAuthInfoMap
+	authErrHandler.refreshActiveMap = refreshActiveMap
+	authErrHandler.disableDestActiveMap = disableDestActiveMap
```

Comment:
What is the reason for initializing those maps and mutexes as global scope variables and then passing them in the handler?

Can just define them inside here and avoid globals?

Reply:
I think I referred this from somewhere but I don't remember now. I will change it.

Reply:
Done